test/helpers/foundry/helpers.mts

```diff
@@ -0,0 +1,393 @@
+  Agent as HttpAgent,
+  request as httpRequest,
+  type IncomingMessage,
+} from 'node:http';
+import { Agent as HttpsAgent, request as httpsRequest } from 'node:https';
+import { join, basename, extname, relative } from 'node:path';
+import { argv, stdout } from 'node:process';
+import { arch, platform } from 'node:os';
+import { Stream } from 'node:stream';
+import { pipeline } from 'node:stream/promises';
+import { extract as extractTar } from 'tar';
+import { Source, Open as Unzip } from 'unzipper';
+import { InferredOptionTypes } from 'yargs';
+import yargs from 'yargs/yargs';
+
+export function noop() {}
+
+export enum BinFormat {
+  Zip = 'zip',
+  Tar = 'tar.gz',
+}
+
+export enum Platform {
+  Windows = 'win32',
+  Linux = 'linux',
+  Mac = 'darwin',
+}
+
+export enum Binary {
+  Anvil = 'anvil',
+  Forge = 'forge',
+  Cast = 'cast',
+  Chisel = 'chisel',
+}
+
+type Options = ReturnType<typeof getOptions>;
+type OptionsKeys = keyof Options;
+type ParsedOptions = {
+  [key in OptionsKeys]: InferredOptionTypes<Options>[key];
+};
+
+export function parseArgs(args: string[] = argv.slice(2)) {
+  const { $0, _, ...parsed } = yargs()
+    // Ensure unrecognized commands/options are reported as errors.
+    .strict()
+    // disable yargs's version, as it doesn't make sense here
+    .version(false)
+    // use the scriptName in `--help` output
+    .scriptName('yarn foundryup')
+    // wrap output at a maximum of 120 characters or `stdout.columns`
+    .wrap(Math.min(120, stdout.columns))
+    .parserConfiguration({
+      'strip-aliased': true,
+      'strip-dashed': true,
+    })
+    // enable ENV parsing, which allows the user to specify foundryup options
+    // via environment variables prefixed with `FOUNDRYUP_`
+    .env('FOUNDRYUP')
+    .command(['$0', 'install'], 'Install foundry binaries', getOptions())
+    .command('cache', '', (yargs) => {
+      yargs.command('clean', 'Remove the shared cache files').demandCommand();
+    })
+    .parseSync(args);
+
+  const command = _.join(' ');
+  switch (command) {
+    case 'cache clean':
+      return {
+        command,
+      } as const;
+    case '':
+    case 'install':
+      return {
+        command: 'install',
+        options: parsed as ParsedOptions,
+      } as const;
+  }
+  throw new Error(`Unknown command: '${command}'`);
+}
+
+function getOptions() {
+  return {
+    binaries: {
+      alias: 'b',
+      type: 'array' as const,
+      multiple: true,
+      description: 'Specify the binaries to install',
+      default: [Binary.Anvil],
+      choices: Object.values(Binary) as Binary[],
+      coerce: (values: Binary[]): Binary[] => [...new Set(values)], // Remove duplicates
+    },
+    repo: {
+      alias: 'r',
+      description: 'Specify the repository',
+      default: 'foundry-rs/foundry',
+    },
+    version: {
+      alias: 'v',
+      description:
+        'Specify the version (see: https://github.com/foundry-rs/foundry/tags)',
+      default: 'nightly',
```

Comment:
should we default to nightly, or should we set a fixed version and update it in separate PRs periodically, to make sure nothing is broken with a version update?

Reply:
The install script I added in `package.json` uses a pinned version. This script defaults to nightly; eventually I think we'll to release this code to npm as a stand-alone package for other developers to use in the future, so I think `nightly` would make sense when we do that, so we should keep it as nightly here. Just my opinion.

Reply:
oh I see, it make sense, thank you for clarifying :pray: